src/scheduler/helpers.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  Audit,
+  ListAnnualReviewPostReminderType,
+  ListEventJsonData,
+  ListItemAnnualReviewProviderReminderType,
+} from "server/models/types";
+import { logger } from "server/services/logger";
+import { ListItemWithHistory } from "server/components/dashboard/listsItems/types";
+import { isLocalHost, SERVICE_DOMAIN } from "server/config";
+
+const defaultTodayDateString = new Date().toLocaleString("default", { year: "numeric", month: "long", day: "numeric" });
+export const now = new Date(Date.now());
+const todayDateString = new Date(Date.UTC(now.getFullYear(), now.getMonth(), now.getDate()));
+
+export function parseDate(dateString = defaultTodayDateString) {
+  const date = new Date(Date.parse(dateString));
+  return new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()));
+}
+
+export function formatDate(date: Date = todayDateString) {
+  const options: Intl.DateTimeFormatOptions = { day: "2-digit", month: "short", year: "numeric" };
+  return date.toLocaleString('en-gb', options);
+}
+
+export function createAnnualReviewProviderUrl(listItem: ListItemWithHistory): string {
+  const protocol = isLocalHost ? "http" : "https";
+  const host = `${protocol}://${SERVICE_DOMAIN}`;
+  const path = `/${listItem.listId}/items/${listItem.id}/annual-review`;
+
+  return `${host}${path}`;
+}
+
+export function isEmailSentBefore(
+  audit: Audit | undefined,
+  reminderType: ListAnnualReviewPostReminderType | ListItemAnnualReviewProviderReminderType
+): boolean {
+  const subsequentEmailsForReminderType = {
+    sendOneMonthPostEmail: [
+      "sendOneMonthPostEmail",
+      "sendOneWeekPostEmail",
+      "sendOneDayPostEmail",
+      "sendStartPostEmail",
+    ],
+    sendOneWeekPostEmail: ["sendOneWeekPostEmail", "sendOneDayPostEmail", "sendStartedPostEmail"],
+    sendOneDayPostEmail: ["sendOneDayPostEmail", "sendStartedPostEmail"],
+    sendStartedPostEmail: ["sendStartedPostEmail"],
+    sendStartedProviderEmail: ["sendStartedProviderEmail"],
+  };
+  if (audit) {
+    if (audit?.jsonData) {
```

Comment:
`if(!audit) { return }`
then the rest of the code/tsc should unwrap `audit` so it's not nullish anymore (I.e no need for `audit?.jsonData`)

probably the perfect example to use guard clauses instead of nested ifs.. 😉 

does tsc not complain about this? 

It doesn't look like all code paths return a boolean. False is only returned if !audit. You could be "trapped" between l50-60. see suggestion below.